pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,720 @@
+from pyaedt.generic.general_methods import generate_unique_name
+from pyaedt.generic.general_methods import pyaedt_function_handler
+
+
+class BaseSimulationSetup(object):
+    """Provide base simulation setup.
+
+    Parameters
+    ----------
+    pedb : :class:`pyaedt.edb.Edb`
+        Inherited AEDT object.
+    edb_object : :class:`Ansys.Ansoft.Edb.Utility.SIWaveSimulationSetup`,
+    :class:`Ansys.Ansoft.Edb.Utility.SIWDCIRSimulationSettings`,
+    :class:`Ansys.Ansoft.Edb.Utility.HFSSSimulationSettings`
+        Edb object.
+    """
+
+    def __init__(self, pedb, edb_setup=None):
+        self._pedb = pedb
+        self._edb_object = edb_setup
+        self._setup_type = ""
+        self._setup_type_mapping = {
+            "kHFSS": self._pedb.simsetupdata.HFSSSimulationSettings,
+            "kPEM": None,
+            "kSIwave": self._pedb.simsetupdata.SIwave.SIWSimulationSettings,
+            "kLNA": None,
+            "kTransient": None,
+            "kQEye": None,
+            "kVEye": None,
+            "kAMI": None,
+            "kAnalysisOption": None,
+            "kSIwaveDCIR": self._pedb.simsetupdata.SIwave.SIWDCIRSimulationSettings,
+            "kSIwaveEMI": None,
+            "kHFSSPI": None,
+            "kDDRwizard": None,
+            "kQ3D": None,
+            "kNumSetupTypes": None,
+        }
+        if self._edb_object:
+            self._name = self._edb_object.GetName()
+
+        self._sweep_list = {}
+
+    @pyaedt_function_handler
+    def _create(self, name=None):
+        """Create a setup."""
```

Comment:
```suggestion
        """Create a simulation setup."""
```
Add a Parameters section and describe the ``name" parameter and what behavior occurs when the default of ``None`` is used.